hooks/post-checkout

```
@@ -1,112 +1,40 @@
-INPUT_VERSION_FILE="${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE-}"
+# Check out the resolved version
+git -C "${INPUT_FOLDER}" checkout "${RESOLVED_VERSION}"
 
-echo "--- Print the plugin inputs"
-get_unresolved_version
-echo "INFO: The Buildkite config version (unresolved) is: ${UNRESOLVED_VERSION:?}"
-echo "INFO: The Buildkite config files will be downloaded from the following repository: ${INPUT_EXTERNAL_REPO:?}"
-echo "INFO: The Buildkite config files will be downloaded into the following folder: ${INPUT_FOLDER:?}"
+# Report the values via environment variables
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="${RESOLVED_VERSION}"
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="${INPUT_REPO_URL}"
+export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${INPUT_FOLDER}"
 
-echo "--- Print the commit in the primary repository"
-git log -n 1
-
-echo "--- Clone from ${INPUT_EXTERNAL_REPO:?} into ${INPUT_FOLDER:?}"
-
-echo "before git clone"
-git clone ${INPUT_EXTERNAL_REPO:?} ${INPUT_FOLDER:?}
-echo "after git clone"
-
-echo "--- Check out ${UNRESOLVED_VERSION:?}"
-PREVIOUS_WORKING_DIRECTORY=`pwd`
-cd ${INPUT_FOLDER:?}
-git checkout ${UNRESOLVED_VERSION:?}
-
-echo "--- Print the commit in the external repository"
-git log -n 1
-RESOLVED_VERSION=`git rev-parse HEAD | tr -s ' '`
-cd ${PREVIOUS_WORKING_DIRECTORY:?}
-
-echo "-- Print the plugin outputs"
-
-export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="${RESOLVED_VERSION:?}"
-export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_EXTERNAL_REPO="${INPUT_EXTERNAL_REPO:?}"
-export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${INPUT_FOLDER:?}"
-
-echo "INFO: The Buildkite config version (unresolved) was: ${UNRESOLVED_VERSION:?}"
-echo "INFO: The Buildkite config version (resolved) is:    ${RESOLVED_VERSION:?}"
-
-### End the main part of the script
+# Also set them via metadata, if we have it available
+if which buildkite-agent >/dev/null 2>/dev/null; then
```

In production, we want to immediately throw an error if the Buildkite agent is not available.

So I would say, let's change this line to something like this:

```suggestion
 if [["${THIS_IS_EXT_BK_PLUGIN_TESTS}" == "true"]; then
```

(Change `THIS_IS_EXT_BK_PLUGIN_TESTS` to a more sane name.)

And then set that environment variable when we run the plugin tests.

---

Yeah, I had only added this before I wrote the fake buildkite agent.  I'll just remove this check.